### PR TITLE
Update to aiida 1.5

### PR DESCRIPTION
**Pull request by danieleongari**
I "modernised" the examples, README and some part of the code, 
still remain.a few issues:

1 - The parser is failing at:
```
  File "/home/daniele/dev_aiida/aiida-qeq/aiida_qeq/parsers/eqeq.py", line 63, in parse
    file=output_folder.open(fname), parse_policy='lazy')
TypeError: a bytes-like object is required, not 'str'
```
The same command also fails in the parser of qeq:
```
  File "/home/daniele/dev_aiida/aiida-qeq/aiida_qeq/parsers/qeq.py", line 63, in parse
    file=output_folder.open(fname), parse_policy='lazy')
```

@ltalirz can you please give a look? You have more expertise on this compatibility problems: last version of the plugin was written for aiida-core-v1.0.0b.

2 - Another related problem on Qeq I just discovered, is related to to its fragile parsing of CIF files: https://github.com/danieleongari/egulp/issues/6
@ltalirz do you think it is better to work on the plugin or the egulp code to fix it? It would likely fail in larger work chains unless we use this specific formatting for the CifData.

3 - I saw that both the parser are storing all the files as SinglefileData
```
            else:
                # add as singlefile
                node = SinglefileData(file=output_folder.open(fname))
                self.out('{}_with_charges'.format(ext), node)
```
I think this is unnecessary and can be discarded, as Qeq/EQeq will be typically used for HT screening and we want to remain as light as possible. The most important file is the .log where in both cases the output that appears on screen is dumped. This file is quite verbose for qeq/egulp (https://github.com/danieleongari/egulp/blob/master/example/HKUST1/output/egulp.log) and shorter for EQeq and it is not very informative: both Qeq/EQeq can fail for some numerical problem, but it is not easy to force the convergence nor easier if one looks at the output file. Therefore I would discard the storage of the log too.

### CODE REVIEW

File: examples/run_qeq.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import aiida_qeq.tests as tests
+import aiida_qeq.data.qeq as data
+from aiida_qeq.data import DATA_DIR
+from aiida.plugins import DataFactory, CalculationFactory
+from aiida.engine import run
+
+QeqCalc = CalculationFactory('qeq.qeq')
+CifData = DataFactory('cif')
+SinglefileData = DataFactory('singlefile')
+Parameters = DataFactory('qeq.qeq')
```
Comment (ltalirz) [inline examples/run_qeq.py:19]: unused

**Comment by ltalirz** [comment]
Re 1: you might need to open with mode 'rb' ? If that doesn't work, I can have a look

Re 2: Better to fix it in egulp

Re 3: No problem with making the retention of outputs lighter - not storing the logs simply means you won't be able to figure out whether there was a problem after the fact.

For users  who use of the `Parameters` class, it would be very easy to add a switch for this - e.g. `store` with options `...` (what is default?), `logs` (default+log)  and `all` (as it is now).

**Comment by danieleongari** [comment]
> not storing the logs
I did not specify that the bulk of the outputs are the same info as in the output CIF but in many other formats. These are totally useless.
As for not storing the .log  this is not necessarily a problem:
* I typically use `gotocomputer` to check what happened as checking the retrieved files is very time consuming
* failed calculations are probably not storing the output because they fail in the process of the parsing: it is anyhow better to check with `gotocomputer`

**Comment by ltalirz** [comment]
Right - up to you; you can either add a field to make the stored outputs configurable (defaulting to a more minimal output set) or you can hardcode it for the moment.

**Comment by danieleongari** [comment]
Examples running, still a couple of problems:

*  deprecation warnings
```
/home/daniele/dev_aiida/aiida-core/aiida/orm/nodes/node.py:53: AiidaDeprecationWarning: `read` used without context manager for <FolderData: uuid: d89dfd46-6975-4c9b-a8ed-f4644c0057c7 (pk: 78561)>. This will raise starting from `aiida-core==2.0.0`
  warnings.warn(msg, AiidaDeprecationWarning)  # pylint: disable=no-member
/home/daniele/dev_aiida/aiida-core/aiida/orm/nodes/node.py:53: AiidaDeprecationWarning: `del` used without context manager for <FolderData: uuid: d89dfd46-6975-4c9b-a8ed-f4644c0057c7 (pk: 78561)>. This will raise starting from `aiida-core==2.0.0`
  warnings.warn(msg, AiidaDeprecationWarning)  # pylint: disable=no-member
```

*  pylint
```
************* Module aiida_qeq.calculations.test_eqeq
aiida_qeq/calculations/test_eqeq.py:15:4: C0103: Variable name "EqeqCalculation" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/calculations/test_eqeq.py:18:4: C0103: Variable name "EQeqParameters" doesn't conform to snake_case naming style (invalid-name)
************* Module aiida_qeq.tests
aiida_qeq/tests/__init__.py:27:4: E0611: No name 'profile' in module 'aiida.backends' (no-name-in-module)
aiida_qeq/tests/__init__.py:27:4: E0401: Unable to import 'aiida.backends.profile' (import-error)
aiida_qeq/tests/__init__.py:100:8: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)

------------------------------------------------------------------
Your code has been rated at 8.62/10 (previous run: 8.62/10, +0.00)

************* Module check_version
.travis-data/check_version.py:14:0: C0103: Constant name "setup_fname" doesn't conform to UPPER_CASE naming style (invalid-name)
.travis-data/check_version.py:22:0: C0103: Constant name "version" doesn't conform to UPPER_CASE naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 9.71/10 (previous run: 9.71/10, +0.00)

************* Module aiida_qeq.data.qeq
aiida_qeq/data/qeq.py:46:0: R0901: Too many ancestors (8/7) (too-many-ancestors)
aiida_qeq/data/qeq.py:66:12: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
aiida_qeq/data/qeq.py:72:12: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
aiida_qeq/data/qeq.py:74:4: R0201: Method could be a function (no-self-use)
aiida_qeq/data/qeq.py:78:4: R0201: Method could be a function (no-self-use)
aiida_qeq/data/qeq.py:123:8: C0103: Variable name "br" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/qeq.py:125:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/qeq.py:126:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/qeq.py:127:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/qeq.py:138:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/qeq.py:139:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/qeq.py:140:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/qeq.py:141:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/qeq.py:142:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/qeq.py:143:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/qeq.py:144:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/qeq.py:12:0: C0411: third party import "from voluptuous import Schema, Optional, Any, ExactSequence" should be placed before "from aiida.orm import Dict" (wrong-import-order)
aiida_qeq/data/qeq.py:13:0: C0411: standard import "from collections import OrderedDict" should be placed before "from voluptuous import Schema, Optional, Any, ExactSequence" (wrong-import-order)
************* Module setup
setup.py:1:0: C0114: Missing module docstring (missing-module-docstring)
setup.py:5:0: C0411: standard import "import json" should be placed before "from setuptools import setup, find_packages" (wrong-import-order)
************* Module conf
docs/source/conf.py:1:0: C0114: Missing module docstring (missing-module-docstring)
docs/source/conf.py:35:4: E0611: No name 'settings' in module 'aiida.backends' (no-name-in-module)
docs/source/conf.py:45:8: C0103: Constant name "html_theme" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:50:4: E0611: No name 'settings' in module 'aiida.backends' (no-name-in-module)
docs/source/conf.py:54:4: E0611: No name 'load_dbenv' in module 'aiida.backends.utils' (no-name-in-module)
docs/source/conf.py:54:4: E0611: No name 'is_dbenv_loaded' in module 'aiida.backends.utils' (no-name-in-module)
docs/source/conf.py:61:0: C0103: Constant name "needs_sphinx" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:84:0: C0103: Constant name "source_suffix" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:91:0: C0103: Constant name "master_doc" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:94:0: C0103: Constant name "project" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:95:0: C0103: Constant name "copyright_first_year" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:96:0: C0103: Constant name "copyright_owners" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:98:0: C0103: Constant name "current_year" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:99:0: C0103: Constant name "copyright_year_string" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:102:0: C0103: Constant name "copyright" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:110:0: C0103: Constant name "release" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:112:0: C0103: Constant name "version" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:119:0: C0103: Constant name "language" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:145:0: C0103: Constant name "show_authors" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:148:0: C0103: Constant name "pygments_style" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:226:0: C0103: Constant name "html_show_sourcelink" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:237:0: C0103: Constant name "html_use_opensearch" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:246:0: C0103: Constant name "html_search_language" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:257:0: C0103: Constant name "htmlhelp_basename" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/source/conf.py:341:0: C0116: Missing function or method docstring (missing-function-docstring)
docs/source/conf.py:50:4: C0412: Imports from package aiida are not grouped (ungrouped-imports)

------------------------------------------------------------------
Your code has been rated at 4.66/10 (previous run: 4.66/10, +0.00)

************* Module aiida_qeq.data.eqeq
aiida_qeq/data/eqeq.py:37:0: R0901: Too many ancestors (8/7) (too-many-ancestors)
aiida_qeq/data/eqeq.py:57:12: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
aiida_qeq/data/eqeq.py:61:12: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
aiida_qeq/data/eqeq.py:63:4: R0201: Method could be a function (no-self-use)
aiida_qeq/data/eqeq.py:11:0: C0411: third party import "from voluptuous import Schema, Optional, Any" should be placed before "from aiida.orm import Dict" (wrong-import-order)
aiida_qeq/data/eqeq.py:12:0: C0411: standard import "from collections import OrderedDict" should be placed before "from voluptuous import Schema, Optional, Any" (wrong-import-order)

------------------------------------------------------------------
Your code has been rated at 9.27/10 (previous run: 9.27/10, +0.00)

************* Module aiida_qeq.data.test_data
aiida_qeq/data/test_data.py:1:0: C0114: Missing module docstring (missing-module-docstring)
aiida_qeq/data/test_data.py:5:0: C0115: Missing class docstring (missing-class-docstring)
aiida_qeq/data/test_data.py:7:4: C0116: Missing function or method docstring (missing-function-docstring)
aiida_qeq/data/test_data.py:9:8: C0103: Variable name "EQeqParameters" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/test_data.py:11:8: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/test_data.py:12:8: C0103: Variable name "p" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/test_data.py:25:8: C0103: Variable name "EQeqParameters" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/test_data.py:27:8: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/test_data.py:33:0: C0115: Missing class docstring (missing-class-docstring)
aiida_qeq/data/test_data.py:35:4: C0116: Missing function or method docstring (missing-function-docstring)
aiida_qeq/data/test_data.py:37:8: C0103: Variable name "QeqParameters" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/test_data.py:39:8: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/test_data.py:40:8: C0103: Variable name "p" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/test_data.py:50:8: C0103: Variable name "QeqParameters" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/test_data.py:52:8: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)
aiida_qeq/data/test_data.py:53:8: C0103: Variable name "p" doesn't conform to snake_case naming style (invalid-name)
************* Module aiida_qeq.calculations.test_qeq
aiida_qeq/calculations/test_qeq.py:14:4: C0103: Variable name "EqeqCalculation" doesn't conform to snake_case naming style (invalid-name)
************* Module conftest
conftest.py:19:4: E0611: No name 'profile' in module 'aiida.backends' (no-name-in-module)
conftest.py:19:4: E0401: Unable to import 'aiida.backends.profile' (import-error)
conftest.py:21:4: R1705: Unnecessary "elif" after "return" (no-else-return)
conftest.py:37:19: W0621: Redefining name 'aiida_profile' from outer scope (line 30) (redefined-outer-name)
conftest.py:52:16: W0621: Redefining name 'aiida_profile' from outer scope (line 30) (redefined-outer-name)
conftest.py:52:0: C0116: Missing function or method docstring (missing-function-docstring)
conftest.py:55:4: C0103: Variable name "SinglefileData" doesn't conform to snake_case naming style (invalid-name)
conftest.py:61:20: W0621: Redefining name 'aiida_profile' from outer scope (line 30) (redefined-outer-name)
conftest.py:61:0: C0116: Missing function or method docstring (missing-function-docstring)
conftest.py:64:4: C0103: Variable name "SinglefileData" doesn't conform to snake_case naming style (invalid-name)
conftest.py:70:15: W0621: Redefining name 'aiida_profile' from outer scope (line 30) (redefined-outer-name)
conftest.py:70:0: C0116: Missing function or method docstring (missing-function-docstring)
conftest.py:73:4: C0103: Variable name "CifData" doesn't conform to snake_case naming style (invalid-name)
conftest.py:79:12: W0621: Redefining name 'aiida_profile' from outer scope (line 30) (redefined-outer-name)
conftest.py:79:0: C0103: Function name "MgO_cif" doesn't conform to snake_case naming style (invalid-name)
conftest.py:79:0: C0116: Missing function or method docstring (missing-function-docstring)
conftest.py:82:4: C0103: Variable name "CifData" doesn't conform to snake_case naming style (invalid-name)
conftest.py:88:19: W0621: Redefining name 'aiida_profile' from outer scope (line 30) (redefined-outer-name)
conftest.py:92:4: C0103: Variable name "SinglefileData" doesn't conform to snake_case naming style (invalid-name)
conftest.py:98:0: C0116: Missing function or method docstring (missing-function-docstring)
conftest.py:9:0: C0411: standard import "import os" should be placed before "import pytest" (wrong-import-order)

------------------------------------------------------------------
Your code has been rated at 5.86/10 (previous run: 5.86/10, +0.00)


--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

**Comment by ltalirz** [comment]
to get rid of the deprecation warning, use `with node.open` ...

for pylint, you can disable the naming rules in docs/conf.py and the tests; the remaining requests should be easy to fix

**Pull request merged**